lib/rucio/core/did.py

```diff
@@ -1888,7 +1889,26 @@ def list_child_datasets(
 
 
 @stream_session
-def list_files(scope, name, long=False, *, session: "Session"):
+def bulk_list_files(dids: "DIDListType", long: bool = False, *, session: "Session"):
+    """
+    List file contents of a list of data identifier.
+
+    :param dids:       A list of dids.
+    :param long:       A boolean to choose if more metadata are returned or not.
+    :param session:    The database session in use.
+    """
+    for did in dids:
+        try:
+            for file_ in list_files(scope=did['scope'], name=did['name'], long=long, session=session):
+                file_['parent_scope'] = did['scope']
+                file_['parent_name'] = did['name']
+                yield file_
```

Comment:
I feel like `file_dict` might be a clearer variable name here, but I think this is perfectly fine too